### PR TITLE
Immigration rules for CalWORKS Child Care program

### LINKED ISSUE #3572: Immigration rules for CalWORKS Child Care programs
Currently not implemented but with the addition of the `immigration_status` enables us to model these rules.

 Undocumented and DACA/TPS should be ineligible for those programs

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #3572

### CODE REVIEW

**Comment by MaxGhenis** [comment]
Could you point to a source? If it's in a document we're already sourcing you can mention it here in the PR - just want some paper trail.

**Comment by PavelMakarchuk** [comment]
> Could you point to a source? If it's in a document we're already sourcing you can mention it here in the PR - just want some paper trail.

The official [DPSS](https://epolicy.dpss.lacounty.gov/epolicy/epolicy/server/general/projects_responsive/ePolicyMaster/index.htm?&area=general&type=responsivehelp&ctxid=&project=ePolicyMaster#t=mergedProjects%2FChild%20Care%2FChild_Care%2F1210_8_Regional_Market_Rate_Ceilings%2F1210_8_Regional_Market_Rate_Ceilings.htm%23Referencesbc-11&rhtocid=_3_3_8_10) site is not available right, now, using the following as a [reference](https://ehsd.org/benefits/calworks-cash-aid/calworks-fact-sheet/):
image

File: policyengine_us/variables/gov/states/ca/cdss/tanf/child_care/eligibility/ca_calworks_child_care_immigration_status_eligible.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    defined_for = StateCode.CA
+    reference = "http://epolicy.dpss.lacounty.gov/epolicy/epolicy/server/general/projects_responsive/ePolicyMaster/index.htm?&area=general&type=responsivehelp&ctxid=&project=ePolicyMaster#t=mergedProjects%2FChild%20Care%2FChild_Care%2F1210_Overview%2F1210_Overview.htm%23Backgroundbc-3&rhtocid=_3_3_0_2"
+
+    def formula(spm_unit, period, parameters):
+        person = spm_unit.members
+        head = person("is_tax_unit_head", period)
+        immigration_status = person("immigration_status", period)
+        status = immigration_status.possible_values
+        citizen = immigration_status == status.CITIZEN
+        resident = immigration_status == status.RESIDENT
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ca/cdss/tanf/child_care/eligibility/ca_calworks_child_care_immigration_status_eligible.py:20]: can we change this code to `LEGAL_PERMANENT_RESIDENT`? everyone is a resident

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Is there other info indicating treatment of mixed-status households?

**Comment by PavelMakarchuk** [comment]
> Is there other info indicating treatment of mixed-status households?

can not find any information on that, currently checking for the head and the child immigration status, still cant access the epolicy site

File: policyengine_us/tests/policy/baseline/gov/states/ca/cdss/tanf/child_care/eligibility/ca_calworks_child_care_child_age_eligible.yaml [hunk 2]
```diff
@@ -3,6 +3,7 @@
   input:
     age: 13
     is_disabled: false
+    immigration_status: RESIDENT
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/states/ca/cdss/tanf/child_care/eligibility/ca_calworks_child_care_child_age_eligible.yaml:6]:
```suggestion
    immigration_status: LEGAL_PERMANENT_RESIDENT
```

**Review by MaxGhenis — APPROVED** [review]
Good to merge after this. We can amend it if we hear from SBN/DPSS different info about mixed-status households.

**Pull request merged**